include/envoy/request_id_utils/request_id_utils.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "envoy/common/pure.h"
+#include "envoy/http/header_map.h"
+
+namespace Envoy {
+namespace RequestIDUtils {
+
+enum class TraceStatus { NoTrace, Sampled, Client, Forced };
+
+/**
+ * Abstract request id utilities for getting/setting the request IDs and tracing status of requests
+ */
+class Utilities {
+public:
+  virtual ~Utilities() = default;
+
+  /**
+   * Directly set a request ID into the provided request headers
+   * @param request_headers supplies the incoming request headers for setting a request ID.
+   */
+  virtual void setRequestID(Http::HeaderMap& request_headers) PURE;
```

Comment:
Please add into the documentation how is `setRequestID` different from `ensureRequestID`